components/Modal.js

```diff
@@ -0,0 +1,56 @@
+import { changeData } from '../api/requests.js';
+
+export class Modal {
+  tableContainer = null;
+  modalWindowContainer = null;
+  bgShadow = null;
+
+  constructor(tableContainer) {
+    this.tableContainer = tableContainer;
+  }
+
+  renderModal(desiredObject, rowId) {
+    this.modalWindowContainer = document.createElement('div');
+    this.modalWindowContainer.classList.add('modal-window-container');
+    this.bgShadow = document.createElement('div');
+    this.bgShadow.classList.add('bg-shadow');
+    this.tableContainer.append(this.modalWindowContainer, this.bgShadow);
+    this.bgShadow.addEventListener('click', () => {
+      this.remove();
+    });
+    let changedDataObj = {};
```

Comment:
Since this is an object and you're going to mutate it, we could use `const` instead of `let`
Let's move it to the class property, to be able to have access elsewhere if needed in the class and do not just proxy it